tests/unittest/_torch/sampler/test_torch_sampler.py

```
@@ -1219,17 +1199,26 @@ def maybe_check_no_sync(func: Callable[[], T]) -> T:
-        sample_states = [
-            maybe_check_no_sync(
-                lambda: sampler.sample_async(
-                    scheduled_requests,
-                    model_outputs=model_outputs,
-                    num_context_logits_prefix_sum=[0],
-                    resource_manager=None,  #  only used for tree sampling, which is not tested here
+        with monkeypatch.context() as patcher:
+            # Ensure that internal sampler data structures are set up for all requests
+            # (production code only considers context requests and examines other not mocked
+            # LlmRequest fields)
+            def _mock_filter(self, requests: ScheduledRequests) -> list[LlmRequest]:
```

Ruff (0.15.1)

[warning] Unused function argument: `self`

(ARG001)